client/utils/currency/index.js

```diff
@@ -349,6 +349,19 @@ function trimEndingZeroes( formattedCurrencyAmount = '' ) {
 		.join( ' ' );
 }
 
+export function trimEndingZeroesAndDecimalSeparator(
+	formattedCurrencyAmount = ''
+) {
+	return formattedCurrencyAmount
+		.split( ' ' )
+		.map( ( chunk ) =>
+			endsWith( chunk, '0' )
+				? trimEnd( chunk, '0' ).replace( '.', '' )
+				: chunk
+		)
+		.join( ' ' );
+}
+
```

Comment:
Actually, do we need this at all? Would the `.00` (which is rare?) make the copy for the button or the notice meaningfully worse?

I think maybe we should remove this, keep it simple, always show amount consistently. 

@nagpai I recommend confirming this detail with @rogermattic / design before we commit to the extra formatting logic complexity.

Reply:
>Actually, do we need this at all? Would the .00 (which is rare?) make the copy for the button or the notice meaningfully worse?

The chances of .00 coming may be a little more than we imagine, since the upper limit for instant deposit is $1000. This can be hit by many merchants. On my test site, seeing 1000.00 looked a bit awkward to me. I asked myself, why is it showing .00 when it is not needed? Sharing screenshots here: 

#### With decimal and zeroes
<img width="695" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/c0feb490-d8e5-42f8-99ad-1abc6b0237d4">

#### Trimmed
<img width="749" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/df66fdd2-ca07-46ab-9ac8-288ff735244a">

Reply:
Right, but is the complexity worth it? Put another way, how important is it to fine tune the max case?

Reply:
> Hehehe this is a good training for my decision making :)

I am so glad to see you take it that way 😃 

Thanks for your prompt check as always, @rogermattic 🙏 !

Reply:
I have rolled back the changes, including updating tests - [8316fa1](https://github.com/Automattic/woocommerce-payments/pull/8970/commits/8316fa15f5fa3963c6bae40033c98caba5ad51cf) .